### PR TITLE
improve/expand design docs for editor and plugins

### PR DESCRIPTION
**Pull request by Anton-4**

### CODE REVIEW

File: design/editor/editor-ideas.md [hunk 1]
```diff
@@ -339,24 +345,6 @@ e.g. you have a test `calculate_sum_test` that only uses the function `add`, whe
...
-If we make an integrated terminal, we can automatically add executed commands to this logbook. This plugin could have a publish button so you can produce useful "blogs" for others with minimal effort.
-
-### Inspiration
-
-- [Boop](https://github.com/IvanMathy/Boop) scriptable scratchpad for developers. Contains collection of useful conversions: json formatting, url encoding, encode to base64...
-- [processing](https://processing.org) Interactive editor, dragging left or right with mouse to change values. Instant results.
-- [flowistry](https://github.com/willcrichton/flowistry) easily track all named values in a certain expression throughout your program.
-- Match blocks of assembly with lines of code: [link](https://twitter.com/dgryski/status/1547952259828330498).
-- [livelits](https://hazel.org/build/livelits/) inline interactive plugin blocks
-
```
Comment (Anton-4) [inline design/editor/editor-ideas.md:359]: I moved this to a different file.

File: design/editor/editor-ideas.md [hunk 2]
```diff
@@ -238,7 +238,11 @@ e.g. you have a test `calculate_sum_test` that only uses the function `add`, whe
...
 - Ability to right click error message > "create github issue".
 - Record (locally) all steps performed by editor + logs and make them searchable for user. The user should be able to scrub through these steps.
-
+- clustering of search results (rg like) based on statistics of result; amount of spaces, avg word length, amount of numbers...
+- Search engine to answer questions about your code base, logs, all intermediary values. Examples: search for "xy" in a string, search for xy in a variable name, show all intermediary values of index variable...
+- When a stacktrace is shown in the logs, highlight all lines of code in the editor that are in the stacktrace.
+- Double-click to select all child expressions of parent: [example](https://twitter.com/disconcision/status/1587156531203678208?s=20&t=GySrwPVnMB6rDKFqRuUcBw).
+- We should encourage users to ask editor something in the command window; e.g. "hide all types", "List docs", "switch light mode"... If the questions can not be matched to existing actions, the user should be given the option to share the question with editor developers so the action can be created or mapped to an existing action.
```
Comment (rtfeldman) [inline design/editor/editor-ideas.md:245]: Not a blocker, but I prefer if we say things like "could" over "should" when we're still in the idea stage. 😄

**Review by rtfeldman — APPROVED** [review]
Sweet, thanks @Anton-4! 🎉

File: design/editor/editor-ideas.md (hunk 2 above)
Comment (Anton-4) [reply design/editor/editor-ideas.md:245]: :+1: I'll keep it in mind

**Pull request merged**